### PR TITLE
Fix artf content get route

### PR DESCRIPTION
**Pull request by likawind**
## Describe your changes and why you are making these changes
This PR fixes artf content get route and re-enables its backend integration test

## Related issue number (if any)
ENG-2712

## Loom demo (if any)
Backend integration test should be sufficient.

## Checklist before requesting a review
- [ ] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [ ] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [ ] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [ ] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: integration_tests/backend/test_reads.py [hunk 1]
```diff
@@ -61,22 +61,22 @@ def setup_class(cls):
         cls.client = aqueduct.Client()
         cls.integration = cls.client.resource(name=pytest.integration)
         cls.flows = {
-            "changing_saves": setup_changing_saves(cls.client, pytest.integration),
+            # "changing_saves": setup_changing_saves(cls.client, pytest.integration),
```
Comment (eunice-chan) [inline integration_tests/backend/test_reads.py:64]: Un-comment out the other flows

File: src/golang/cmd/server/handler/v2/node_artifact_result_content_get.go [hunk 2]
```diff
@@ -160,54 +154,8 @@ func (h *NodeArtifactResultContentGetHandler) Perform(ctx context.Context, inter
...
 	}
 
 	return &nodeResultGetResponse{IsDownsampled: isDownsampled, Content: data}, http.StatusOK, nil
 }
-
-// This custom implementation of SendResponse constructs a multipart form response with two fields:
-// 1: "metadata" contains a json serialized blob of artifact result metadata.
-// 2: "data" contains the artifact result data blob generated the serialization method
-// specified in the metadata field.
-func (*NodeArtifactResultContentGetHandler) SendResponse(w http.ResponseWriter, interfaceResp interface{}) {
```
Comment (eunice-chan) [inline src/golang/cmd/server/handler/v2/node_artifact_result_content_get.go:175]: Why is this removed?

File: src/golang/cmd/server/handler/v2/node_artifact_result_content_get.go [hunk 3]
```diff
@@ -106,9 +100,9 @@ func (h *NodeArtifactResultContentGetHandler) Perform(ctx context.Context, inter
 	args := interfaceArgs.(*nodeResultGetArgs)
 	emptyResp := &nodeResultGetResponse{}
 
-	dag, err := h.DAGRepo.GetByDAGResult(
+	dag, err := h.DAGRepo.Get(
 		ctx,
-		args.nodeResultID,
+		args.dagID,
```
Comment (eunice-chan) [inline src/golang/cmd/server/handler/v2/node_artifact_result_content_get.go:105]: Great! Can you also make the change for the other `*_content_get` endpoints as well?

File: integration_tests/backend/test_reads.py [hunk 4]
```diff
@@ -496,29 +496,33 @@ def test_endpoint_node_artifact_get(self):
...
+        artifact_id = str(artifact_ids[0])
+
+        resp = self.get_response(
+            self.GET_NODE_ARTIFACT_RESULTS_TEMPLATE % (flow_id, dag_id, artifact_id)
+        ).json()
+        downstream_ids = [GetArtifactResultResponse(**result).id for result in resp]
+        for downstream_id in downstream_ids:
+            artifact_result_id = str(downstream_id)
+            resp = self.get_response(
+                self.GET_NODE_ARTIFACT_RESULT_CONTENT_TEMPLATE
```
Comment (eunice-chan) [inline integration_tests/backend/test_reads.py:520]: I've been checking the format of the response is correct by checking to see if I could convert them to the *Response class I create for each endpoint e.g. `result = GetNodeArtifactResponse(**resp)` does not error out. Would be useful if you could make a ContentResponse class which can be used for the other ContentGet routes as well

File: integration_tests/backend/test_reads.py [hunk 5]
```diff
@@ -496,29 +496,33 @@ def test_endpoint_node_artifact_get(self):
...
             assert set(all_output_counts) == set([0, 1])
 
-    # def test_endpoint_node_artifact_result_content_get(self):
-    #     flow_id, n_runs = self.flows["flow_with_multiple_operators"]
-    #     flow = self.client.flow(flow_id)
-    #     workflow_resp = flow._get_workflow_resp()
-    #     dag_id = workflow_resp.workflow_dag_results[0].workflow_dag_id
-    #     dag_result_id = workflow_resp.workflow_dag_results[0].id
+    def test_endpoint_node_artifact_result_content_get(self):
+        flow_id, _ = self.flows["flow_with_multiple_operators"]
```
Comment (eunice-chan) [inline integration_tests/backend/test_reads.py:500]: I've been checking for the flows `flow_with_metrics_and_checks` and `flow_with_multiple_operators` (see `test_endpoint_node_artifact_get`). If adding `flow_with_metrics_and_checks` is a straightforward extension, would like it if you add that too.

File: src/golang/cmd/server/handler/v2/node_artifact_result_content_get.go (hunk 2 above)
Comment (likawind) [reply src/golang/cmd/server/handler/v2/node_artifact_result_content_get.go:175]: We save  the data in memory, so there's no need to send the content as a form data. Json is sufficient to handle

**Review by eunice-chan — APPROVED** [review]

File: src/golang/cmd/server/handler/v2/node_metric_result_content_get.go [hunk 6]
```diff
@@ -88,9 +88,9 @@ func (h *NodeMetricResultContentGetHandler) Perform(ctx context.Context, interfa
 	args := interfaceArgs.(*nodeResultGetArgs)
 	emptyResp := &nodeResultGetResponse{}
 
-	dag, err := h.DAGRepo.GetByDAGResult(
```
Comment (likawind) [inline src/golang/cmd/server/handler/v2/node_metric_result_content_get.go:91]: @eunice-chan

File: src/golang/cmd/server/handler/v2/node_metric_result_content_get.go [hunk 7]
```diff
@@ -148,8 +148,6 @@ func (h *NodeMetricResultContentGetHandler) Perform(ctx context.Context, interfa
 			return emptyResp, http.StatusOK, nil
 		}
 
-		return emptyResp, http.StatusInternalServerError, errors.Wrap(err, "Failed to retrieve data for the artifact result.")
```
Comment (likawind) [inline src/golang/cmd/server/handler/v2/node_metric_result_content_get.go:151]: @eunice-chan

File: src/golang/cmd/server/handler/v2/node_check_result_content_get.go [hunk 8]
```diff
@@ -148,8 +148,6 @@ func (h *NodeCheckResultContentGetHandler) Perform(ctx context.Context, interfac
 			return emptyResp, http.StatusOK, nil
 		}
 
-		return emptyResp, http.StatusInternalServerError, errors.Wrap(err, "Failed to retrieve data for the artifact result.")
```
Comment (likawind) [inline src/golang/cmd/server/handler/v2/node_check_result_content_get.go:151]: @eunice-chan

File: src/golang/cmd/server/handler/v2/node_check_result_content_get.go [hunk 9]
```diff
@@ -88,9 +88,9 @@ func (h *NodeCheckResultContentGetHandler) Perform(ctx context.Context, interfac
 	args := interfaceArgs.(*nodeResultGetArgs)
 	emptyResp := &nodeResultGetResponse{}
 
-	dag, err := h.DAGRepo.GetByDAGResult(
+	dag, err := h.DAGRepo.Get(
```
Comment (likawind) [inline src/golang/cmd/server/handler/v2/node_check_result_content_get.go:91]: @eunice-chan
